Automattic/newspack-popups
# fix: prevent insertions which break layout; improve classic-content insertion

### All Submissions:






### Changes proposed in this Pull Request:



Closes #555.

### How to test the changes in this Pull Request:

1. On `master`,
1. Create a post which has mid-content headings and left- or right-aligned (floated) images
2. Concoct a set of inline prompts (by assigning percentage-placement values) that break the layout: 
    1. prompts inserted right after a heading

<img width="700" alt="image" src="https://user-images.githubusercontent.com/7383192/139457498-44bb215e-9ade-48bc-97a3-1fa138e8f164.png">
 
    2. prompts inserted right after a left- or right-aligned image
 
<img width="724" alt="image" src="https://user-images.githubusercontent.com/7383192/139457349-6a4ae6d6-1989-4001-b329-6bc65ec9b26c.png">
  
3. Observe, as described in #555, that prompts can break the layout
4. Switch to this branch, observe issues solved:

<img width="708" alt="image" src="https://user-images.githubusercontent.com/7383192/139457697-6b14096e-a33c-4f3b-afb2-cc4ed249d5a4.png">

<img width="745" alt="image" src="https://user-images.githubusercontent.com/7383192/139457705-42b92799-52cb-4ec6-8e97-620054cecb5d.png">

5. Now do the same, but with a legacy content post. This can be mimicked by creating a post with a single "Classic" block.
6. Observe that prompts are inserted correctly into the classic content.  

### Other information:
